services/app-web/src/pages/Wrapper/AuthenticatedRouteWrapper.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import { Modal } from '../../components/Modal/Modal'
+import { ModalRef, Alert } from '@trussworks/react-uswds'
+import { createRef, useCallback, useEffect } from 'react'
+import { useAuth } from '../../contexts/AuthContext'
+import { AuthModeType } from '../../common-code/config'
+import { extendSession } from '../Auth/cognitoAuth'
+import styles from '../StateSubmission/ReviewSubmit/ReviewSubmit.module.scss'
+import { dayjs } from '../../common-code/dateHelpers/dayjs'
+import { useLDClient } from 'launchdarkly-react-client-sdk'
+import { featureFlags } from '../../common-code/featureFlags'
+
+export const AuthenticatedRouteWrapper = ({
+    children,
+    setAlert,
+    authMode,
+}: {
+    children: React.ReactNode
+    setAlert?: React.Dispatch<React.ReactElement>
+    authMode: AuthModeType
+}): React.ReactElement => {
+    const { logout, isSessionExpiring, timeUntilLogout, updateSessionExpiry } =
+        useAuth()
+    const [announcementSeed] = React.useState<number>(timeUntilLogout)
+    const announcementTimes: number[] = []
+    for (let i = announcementSeed; i > 0; i -= 10) {
+        announcementTimes.push(i)
+    }
+    const ldClient = useLDClient()
+    const minutesUntilExpiration = ldClient?.variation(
+        featureFlags.MINUTES_UNTIL_SESSION_EXPIRES,
+        30
+    )
+    const modalRef = createRef<ModalRef>()
+    const LocalStorage = window.localStorage
+    const logoutSession = useCallback(() => {
+        updateSessionExpiry(false)
+        LocalStorage.removeItem('LOGOUT_TIMER')
+        if (logout) {
+            logout().catch((e) => {
+                console.log('Error with logout: ', e)
+                setAlert &&
+                    setAlert(
+                        <Alert
+                            data-testid="Error400"
+                            style={{ width: '600px', marginBottom: '5px' }}
+                            type="error"
+                            heading="Oops! Something went wrong"
+                        />
+                    )
+            })
+        }
+    }, [logout, setAlert, updateSessionExpiry, LocalStorage])
+
+    const resetSessionTimeout = () => {
+        updateSessionExpiry(false)
+        const sessionExpirationTime = dayjs(Date.now()).add(
+            minutesUntilExpiration,
+            'minute'
+        )
+        try {
+            LocalStorage.setItem(
+                'LOGOUT_TIMER',
+                sessionExpirationTime.toISOString()
+            )
+        } catch (e) {
+            console.log('Error setting logout timer: ', e)
+        }
+        if (authMode !== 'LOCAL') {
+            void extendSession()
+        }
+    }
+
+    useEffect(() => {
+        modalRef.current?.toggleModal(undefined, isSessionExpiring)
+    }, [isSessionExpiring, modalRef])
+
+    useEffect(() => {
+        if (timeUntilLogout < 1) {
```

Comment:
Would you mind sneaking in a `setAlert` here to address https://qmacbis.atlassian.net/browse/MR-53 as well? I think that might address the bug - rather than waiting for the AuthContext refactor I have on another branch

Reply:
I don't think the quick fix is going to work (and didn't work locally) because everything gets wiped out on logout.  We'll have to persist the "just got logged out" state, probably in local storage.  (But open to other ideas.)

Reply:
that makes sense.